src/engine/MaxConstraint.cpp

```diff
@@ -190,18 +194,26 @@ void MaxConstraint::notifyLowerBound( unsigned variable, double value )
 
     if ( isActive() && _boundManager )
     {
+        if ( proofs )
+            for ( const auto &element : _elements )
+                createElementTighteningRow( element );
+
         // TODO: optimize this. Don't need to recompute ALL possible bounds,
         // Can focus only on the newly learned bound and possible consequences.
         List<Tightening> tightenings;
         getEntailedTightenings( tightenings );
-        for ( const auto &tightening : tightenings )
+
+        if ( !proofs )
```

Comment:
Why isn't the part below performed when proofs are on?
Also, currently we call "getEntailedTightenings" and then don't do anything with the returned information in that case, so maybe put that all under the 'if'. In the next function, too.

Reply:
When calling getEntailedTightenings, we have all info necessary for updating the explanations and lemmas (for example, the element that causes a tightening for f), but we don't have it in the wrapping function. I thought it would be better to update the explanation directly inside getEntailedTightenings. and not to reproduce this info twice (in this function and in the next one).
I'm not sure I understand the suggestion. In any case, the calling getEntailedTightenings is a part of the proof production for MaxConstraint.